main.tf

```diff
@@ -1,12 +1,31 @@
 # Kubernetes provider
```

Comment:
Its common practice to have a generic `main.tf` for bits that are generic and not resource specific. This captures the providers, common local/data sources, and the random string resource used in the naming scheme